### ISSUE TITLE
Viewport shifting in ios Safari/Chrome when typing in hosted field

### ISSUE THREAD

**Issue by tdrake-cb** [braintree/braintree-web#363]
### General information

* SDK version: 3.32.1
* Environment: Production And Sandbox
* Browser and OS: Safari 11 and ios 11.2 

### Issue description

I am having an issue with the mobile viewport shifting when typing in a hosted field within ios safari and chrome. When a user selects the hosted field within the iframe, the browser correctly focuses on the input and pulls up the virtual keyboard. If the input field is close to the top of the virtual keyboard, when a user types a number for credit card number or cvv the viewport is shifted downward and moves the input completely out of view.

Based on several stack overflow questions, my guess is that the browser and the iframe are not interacting correctly. Others have suggested that adding the flowing css within the iframe is a potential fix, not sure of the potential side effects...

```
html,
body {
  height: 100%;
  -webkit-overflow-scrolling: touch;
  overflow: auto;
}
````

Can anyone else reproduce this?

**Comment by crookedneighbor** [comment]
This is happening because of a bug with how iOS Safari maintains focus in inputs inside of iframes. We've tried numerous workaround with little to no success, but we'll give your snippet a try as well.

**Comment by alovak** [comment]
Not sure, but I was able to fix it with setting focus (and then blur) on fake field inside parent page close to iframe.

@tdrake-cb you can try to reproduce issue on my page (it's not Braintree Hosted fields, it's a bit different thing) [here](http://js.powerpayments.tech/). If it works, then we can discuss with @crookedneighbor how it works on my side :)

**Comment by ThoamsClark** [comment]
Any update on this? I have users complaining of the exact same symptoms (primarily iPad users, if that helps at all).

So far I haven't been able to reproduce the issue myself which has been very frustrating. Could anyone give me a reliable way to do so?

**Comment by ThoamsClark** [comment]
I have been able to replicate using Xcode's simulator running iOS 11.2 on various iPads. I wasn't able to reproduce it on mobiles devices.

Currently my workaround is to detect if the containing element has moved and move it back. This is less than ideal.

**Comment by crookedneighbor** [comment]
No real update on this. We're still looking at it though.

**Comment by crookedneighbor** [comment]
We've just released some changes in v3.78.2 that address bugs with iframes in iOS Safari/Chrome. Can you check and see if you're still having issues with this?

**Comment by crookedneighbor** [comment]
Closing this due to inactivity. Feel free to comment here if you're still having issues.
